### PR TITLE
We do a check against a TransformedRecord's self defined is_valid pro…

### PR DESCRIPTION
**Pull request by CuriousG102**
…perty here, and if is_valid is False, we bail out and return None. None gets appended to the list of records to return which subsequently crashes import_file in import_salary_data when it is encountered.

Thankfully, this never happens, because all of our transform functions don't generate lists with invalid records. They access the same property to determine whether to append the return value from record's as_dict method in the first place. Which renders the check in as_dict redundant and irrelevant.

Summing up, this code never runs. If it did run, it would break everything. Taking it out will allow changes that follow to modify code upstream and downstream without a nasty and hard to trace surprise.
#99

### CODE REVIEW

**Comment by CuriousG102** [comment]
If we want to keep the behavior of checking for validity before returning a dict, we could instead have a line
assert self.is_valid . Then the failure in the person's code will be obvious for them rather than a strange silent failure down the line.

**Comment by CuriousG102** [comment]
@rdmurphy

**Comment by rdmurphy** [comment]
We should chat about this in person. I think I get what you mean, but I'm having a little trouble following.

**Comment by rdmurphy** [comment]
:+1: Good catch.
